### ISSUE TITLE
Can not close or mention issues via commit

### ISSUE THREAD

**Issue by BANG88** [go-gitea/gitea#3848]
- Gitea version (or commit ref): 1.4.0
- Git version: latest
- Operating system: Windows
- Database (use `[x]`):
  - [ ] PostgreSQL
  - [ ] MySQL
  - [ ] MSSQL
  - [x] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [ ] Yes (provide example URL)
  - [x] No https://try.gitea.io/testx/test
  - [ ] Not relevant
- Log gist:

## Description

I am upgrade from gogs to gitea last month, everything looks fine ATM. but today I found I can not close issue via commit or make reference.

But the link `#1` is correct can jump to the issue 1.

## Screenshots

<img width="1149" alt="screenshot at apr 25 17-37-13" src="https://user-images.githubusercontent.com/433259/39237872-5549882e-48af-11e8-980c-19c093997793.png">
<img width="1183" alt="screenshot at apr 25 17-36-37" src="https://user-images.githubusercontent.com/433259/39237873-5648aa70-48af-11e8-8c9a-000546809fe5.png">

**Comment by thehowl** [comment]
This seems weird. Do the logs say anything?

**Comment by BANG88** [comment]
@thehowl no logs about this issue. does this needs some git-hooks let it work or something else.

**Comment by thehowl** [comment]
No, this should work out of the box. Was the issue actually created before you made the attempt to close it?

On 27 April 2018 03:03:42 CEST, Qibang <notifications@github.com> wrote:
>@thehowl no logs about this issue. does this needs some git-hooks let
>it work or something else.
>
>-- 
>You are receiving this because you were mentioned.
>Reply to this email directly or view it on GitHub:
>https://github.com/go-gitea/gitea/issues/3848#issuecomment-384832813

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Comment by BANG88** [comment]
Yes it is. I have create 140 issues before and this happened in the other repo too.

**Comment by krrutkow** [comment]
To hopefully clarify a bit, commits should open/close issues when directly pushed to a repo, but do not affect issues if they were merged into a repo from a pull request.

Once merged, #3695 will hook up the latter functionality.

**Comment by BANG88** [comment]
@thehowl Hows going. We need this working as expected. If I can help out please tell me what should I do. 

Thanks

**Comment by thehowl** [comment]
No, I still haven't quite understood why this is happening. Can you perhaps link us the repository where this happens?

@krrutkow This does not seem to be the case; as you can see from the screenshot on top, there is no merge commit, so it's not a merge from a pull request.

**Comment by BANG88** [comment]
@thehowl the repo are private so can not access from public network

**Comment by krrutkow** [comment]
Another possibility is that gitea's git hooks are not properly set up for the repository.  If they are not present or if they lack the proper read+execute permissions they will not trigger events in gitea when pushes are made to the repository.

**Comment by BANG88** [comment]
update git hooks now works

**Comment by pcopissa** [comment]
I think I observe the same symptoms (but maybe I have the wrong expectations about how the system works ?)
Here are the steps I do (on Gitea 1.5.2):
Create an issue (#1 in my case...). Issue shows as "open".
Added + committed + pushed some file with a comment "`fixes #1`" into `master`.
I expected to see the issue change to "close" but it didn't not. I tried quite a few variant (`fixes: #1`, `Fixes #1`, `Fixes: #1`, `fix #1` and countless others and none worked as I expected) 
Is closing an issue from a commit / push possible ? If so how ?
